chatlas/_chat.py

```diff
@@ -807,6 +807,39 @@ def console(
             self.chat(user_input, echo=echo, stream=stream, kwargs=kwargs)
             print("")
 
+    def to_solver(self):
+        """Return an InspectAI solver that proxies prompts through this chat."""
+
+        try:
+            from inspect_ai.model import ChatMessageAssistant, ModelOutput
+            from inspect_ai.solver import solver
+        except ImportError as exc:  # pragma: no cover - optional dependency
+            raise ImportError(
+                "Chat.to_solver() requires the optional dependency `inspect-ai`. "
+                "Install it with `pip install inspect-ai`."
+            ) from exc
+
+        chat_instance = self
+
+        @solver
+        def _solver(_chat: "Chat"):
+            async def solve(state, generate):
+                user_prompt = getattr(state.user_prompt, "text", None)
+                if user_prompt is None:
+                    user_prompt = str(state.user_prompt)
+
+                response = await wrap_async(chat_instance.chat)(user_prompt)
+                content = response if isinstance(response, str) else str(response)
+
+                assistant_message = ChatMessageAssistant(content=content)
+                state.messages.append(assistant_message)
```

Comment:
Assuming we want to respect the conversation history (which I think we do), seems we'll also need a way to translate (and append) `self.get_turns()` into `state.messages`. That might be a bit tricky, but hopefully at least translating `self.system_prompt` is straight-forward?